lib/bridge.js

```diff
@@ -17,51 +18,111 @@ var Callback = require('./callback');
  */
 var Bridge = function Bridge (ip, port, token, options) {
   if (!(this instanceof Bridge)) {
-    return new Bridge(ip, port, token);
+    return new Bridge(ip, port, token, options);
   }
 
   this.ip = ip;
   this.port = parseInt(port, 10);
   this.token = token;
+  this.type = 1; // 1 = hardware, 2 = software
+  
   this.delayBetweenRequests = options && options.delayBetweenRequests || 250;
   this.delayer = Promise.resolve();
 
   if (!this.ip || !this.port || !this.token || this.port < 1 || this.port > 65536) {
     throw new Error('Please check the arguments!');
   }
+  
+  // get bridge type
+  this._getBridgeType();
+};
+
+/**
+ * This function retrieves the type of bridge (1 = hardware, 2 = software).
+ *
+ * @param void
+ * @returns void
+ * @private
+ */
+Bridge.prototype._getBridgeType = function _getBridgeType () {
+
+	var self = this;
+	self.delayer = new Promise(resolve =>
+	{
+		self._request('info')
+			.then(bridge =>
+			{
+				console.log(JSON.stringify(bridge));
+				self.type = bridge.bridgeType;
+			})
+			.catch(err => self.type = 2)
```

Comment:
Please remove the debug logging in the good case and maybe add an error logging (console.error) in the catch, since this should not really happen, because the software bridge understands the info-request, right?

Reply:
The software bridge will go through the `catch` block and thus the type will be set to 2.